src/system/game-data.ts

```diff
@@ -553,6 +570,84 @@ export class GameData {
     });
   }
 
+  public async getRunHistoryData(scene: BattleScene): Promise<Object> {
+    if (!Utils.isLocal) {
+      /**
+       * Networking Code DO NOT DELETE!
+       *
+      const response = await Utils.apiFetch("savedata/runHistory", true);
+      const data = await response.json();
+      */
+      if (localStorage.hasOwnProperty(`runHistoryData_${loggedInUser?.username}`)) {
+        let cachedResponse = localStorage.getItem(`runHistoryData_${loggedInUser?.username}`);
+        if (cachedResponse) {
+          cachedResponse = JSON.parse(decrypt(cachedResponse, true));
+        }
+        const cachedRHData = cachedResponse ?? {};
+        // check to see whether cachedData or serverData is more up-to-date
+        /**
+       * Networking Code DO NOT DELETE!
+       *
+        if ( Object.keys(cachedRHData).length >= Object.keys(data).length ) {
+          return cachedRHData;
+        }
+        */
+        return cachedRHData;
+      } else {
+        localStorage.setItem(`runHistoryData_${loggedInUser?.username}`, JSON.parse(encrypt("", true)));
+        return {};
+      }
+      //return data;
+    } else {
+      let cachedResponse = localStorage.getItem(`runHistoryData_${loggedInUser?.username}`);
+      if (cachedResponse) {
+        cachedResponse = JSON.parse(decrypt(cachedResponse, true));
+      }
+      const cachedRHData = cachedResponse ?? {};
+      return cachedRHData;
+    }
+  }
+
+  async saveRunHistory(scene: BattleScene, runEntry : SessionSaveData, victory: boolean): Promise<boolean> {
+
+    let runHistoryData = await this.getRunHistoryData(scene);
+    if (!runHistoryData) {
+      runHistoryData = {};
+    }
+    const timestamps = Object.keys(runHistoryData);
+    const timestampsNo = timestamps.map(Number);
+
+    // Arbitrary limit of 25 entries per User --> Can increase or decrease
+    if (timestamps.length >= 25) {
+      const oldestTimestamp = Math.min.apply(Math, timestampsNo);
+      delete runHistoryData[oldestTimestamp.toString()];
+    }
+
+    const timestamp = (runEntry.timestamp).toString();
+    runHistoryData[timestamp] = {};
+    runHistoryData[timestamp]["entry"] = runEntry;
+    runHistoryData[timestamp]["victory"] = victory;
+    // Not implemented at the moment, but leaving room for future work
+    runHistoryData[timestamp]["favorite"] = false;
```

Comment:
```suggestion
    runHistoryData[timestamp] = {
      entry: runEntry,
      victory: victory,
    };
```

And you should move the `Not implemented at the moment, but leaving room for future work` into the description of the interface

```ts
export interface RunEntry {
  entry: SessionSaveData;
  victory: boolean;
  /** Not used yet. Potential future feature */
  favorite?: boolean;
}